Incorrect initialization with explicit time dependence

Currently, `time` in `initialAssignment`s is always replaced by 0, independent of the chosen simulation start time. I don't think this makes sense.
While the current SBML spec is not super clear on this topic, some clarification has started [here](https://github.com/sbmlteam/sbml-specifications/pull/408/files). 

```python

def test_t0(tempdir):
    from amici.antimony_import import antimony2amici

    model_name = "test_t0"
    antimony2amici(
        """
        my_time = time
        my_time' = 1
    """,
        model_name=model_name,
        output_dir=tempdir,
    )

    model_module = import_model_module(model_name, tempdir)

    model = model_module.get_model()
    model.setTimepoints([2])
    model.setT0(2)

    solver = model.getSolver()
    rdata = amici.runAmiciSimulation(model, solver)
    assert rdata.x == [[2.0]], rdata.x
```


```python
>       assert rdata.x == [[2.0]], rdata.x
E       AssertionError: array([[0.]])
E       assert array([[0.]]) == [[2.0]]
E         
E         Full diff:
E         + array([[0.]])
E         - [
E         -     [
E         -         2.0,
E         -     ],
E         - ]
```

--- a/python/sdist/amici/sbml_import.py
+++ b/python/sdist/amici/sbml_import.py
@@ -1382,7 +1382,7 @@ class SbmlImporter:
                     ),
                 }
 
-        # Parameters that need to be turned into expressions
+        # Parameters that need to be turned into expressions or species
         #  so far, this concerns parameters with symbolic initial assignments
         #  (those have been skipped above) that are not rate rule targets
         for par in self.sbml.getListOfParameters():
@@ -1391,12 +1391,33 @@ class SbmlImporter:
                 and not ia.is_Number
                 and not self.is_rate_rule_target(par)
             ):
-                self.symbols[SymbolId.EXPRESSION][
-                    _get_identifier_symbol(par)
-                ] = {
-                    "name": par.getName() if par.isSetName() else par.getId(),
-                    "value": ia,
-                }
+                if not ia.has(sbml_time_symbol):
+                    self.symbols[SymbolId.EXPRESSION][
+                        _get_identifier_symbol(par)
+                    ] = {
+                        "name": par.getName()
+                        if par.isSetName()
+                        else par.getId(),
+                        "value": ia,
+                    }
+                else:
+                    # Convert parameters with initial assignments that
+                    #  explicitly depend on time to species.
+                    #  We can't represent that as expression, since the
+                    #  initial simulation time is only known at the time of the
+                    #  simulation, so we can't substitute it.
+                    self.symbols[SymbolId.SPECIES][
+                        _get_identifier_symbol(par)
+                    ] = {
+                        "name": par.getName()
+                        if par.isSetName()
+                        else par.getId(),
+                        "init": ia,
+                        "dt": sp.Float(0),
+                        "amount": True,
+                        "constant": True,
+                        "index": len(self.symbols[SymbolId.SPECIES]),
+                    }
 
     @log_execution_time("processing SBML reactions", logger)
     def _process_reactions(self):
@@ -2342,8 +2363,6 @@ class SbmlImporter:
                 # no need to recurse here, as value is numeric
                 init = sp.Float(element.getValue())
                 sym_math = sym_math.subs(var, init)
-
-        sym_math = smart_subs(sym_math, sbml_time_symbol, sp.Float(0))
 
         sym_math = _dummy_to_rateof(sym_math, rateof_to_dummy)
 

--- a/python/tests/test_sbml_import.py
+++ b/python/tests/test_sbml_import.py
@@ -929,3 +929,30 @@ def test_heaviside_init_values_and_bool_to_float_conversion(tempdir):
 
     assert np.all(rdata.by_id("a") == np.array([2, 2, 2])), rdata.by_id("a")
     assert np.all(rdata.by_id("b") == np.array([0, 1, 1])), rdata.by_id("b")
+
+
+@skip_on_valgrind
+def test_t0(tempdir):
+    """Test that a custom initial time for the simulation is applied correctly
+    during species initialization."""
+    from amici.antimony_import import antimony2amici
+
+    model_name = "test_t0"
+    antimony2amici(
+        """
+        my_time = time
+        my_time' = 1
+    """,
+        model_name=model_name,
+        output_dir=tempdir,
+    )
+
+    model_module = import_model_module(model_name, tempdir)
+
+    model = model_module.get_model()
+    model.setTimepoints([2])
+    model.setT0(2)
+
+    solver = model.getSolver()
+    rdata = amici.runAmiciSimulation(model, solver)
+    assert rdata.x == [[2.0]], rdata.x

